lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2061,6 +2061,51 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
         b.create<arith::ConstantOp>(loc, FloatAttr::get(elementType, 1.0));
     return b.create<arith::DivFOp>(loc, one, payloadArgs[0]);
   }
+  if (auto thresholdOp = dyn_cast<AtenThresholdOp>(op)) {
+    // The approach used here is as follows:
+    //        result = self <= threshold ? value : self
+    AtenThresholdOp::Adaptor adaptor(operands);
+    Type dtype = converter->convertType(thresholdOp.getType())
+                     .cast<RankedTensorType>()
+                     .getElementType();
+
+    Value self = payloadArgs[0];
+    Value threshold = convertScalarToDtype(b, loc, adaptor.threshold(), dtype);
+    Value value = convertScalarToDtype(b, loc, adaptor.value(), dtype);
+
+    Value predicate;
+    if (dtype.isa<mlir::FloatType>())
+      predicate = b.create<arith::CmpFOp>(loc, arith::CmpFPredicate::ULE, self,
+                                          threshold);
+    else
+      predicate = b.create<arith::CmpIOp>(loc, arith::CmpIPredicate::sle, self,
+                                          threshold);
+    return b.create<SelectOp>(loc, predicate, value, self);
+  }
+  if (auto thresholdBackward = dyn_cast<AtenThresholdBackwardOp>(op)) {
+    // The approach used here is as follows:
+    //        result = self <= threshold ? 0 : grad
+    AtenThresholdBackwardOp::Adaptor adaptor(operands);
+    Type dtype = converter->convertType(thresholdBackward.getType())
+                     .cast<RankedTensorType>()
+                     .getElementType();
+
+    Value grad = convertScalarToDtype(b, loc, payloadArgs[0], dtype);
+    Value self = convertScalarToDtype(b, loc, payloadArgs[1], dtype);
+    Value threshold = convertScalarToDtype(b, loc, adaptor.threshold(), dtype);
+    Value constantZero =
+        b.create<arith::ConstantOp>(loc, b.getF32FloatAttr(0.0));
```

Comment:
nit: could just use `b.getZeroAttr(dtype)`

Reply:
Is everything okay, apart from this?

Reply:
yes! LGTM otherwise.